interview-questions.md

```diff
@@ -8,42 +8,42 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: A float is considered imprecise in ruby, however, a decimal is not the same as a float. When dealing with decimals, decimals can return a lot of number. For example: 3.459586053. A float returns smaller numbers. For example: 2.3, 4.23, and so on. Integers are whole numbers. They can also be negative. 
 
-1. Ruby has an implicit return. What does that mean?
+3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: Implicit return means that the return is implied, or expected. Therefore, returns are not required when dealing with ruby. One can add the return, but it is not needed. This is different from an explicit return; an explicit return means that a return is not automatically implied, therefore, we must add the return keyword when dealing with explicit return. This is true in JavaScript; JavaScript does have implicit returns, but they are not as common as Ruby, where every return is implicit.
 
-Researched answer:
+Researched answer: In ruby, returns are implicit because ruby automatically returns the last line of code the user provides. This is the last line before then end keyword. This is why a return keyword is not required in Ruby, however, it can be used. The end result will be exactly the same. 
 
-1. What is a block in Ruby?
+4. What is a block in Ruby?
 
-Your answer:
+Your answer: A block in ruby is an anonymous function that can be used inside of methods in order to execute a specified action. Blocks often require the do and end keywords. That is how we specify the beggining of a block. Blocks can take arguments, we include those arguments inside of pipes, or | |. Blocks can also be written inside of curly brackets {} if they are only a single line of code. There are many blocks in ruby, such as: .map, .filter, .each. 
 
-Researched answer:
+Researched answer: A block is the same thing as a method, but it does not belong to an object. Blocks can accept arguments and return values, however, blocks do not have a name of their own. 
 
-1. How do you extract a value in a Ruby hash?
+5. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: Since hashes in ruby are objects we can extract values in hash by referring to the hash, and the referring to the key inside of that hash. For example: if we have a hash named cars, we can say p cars[:japanese]. Japanese is referring to the key japanese, which will the extract the value that belongs to the key :japanese.
 
-Researched answer:
+Researched answer: Whilst the above statement is true, we can literally extract values from an object's keys. We can use the extract method to extract values that we want in ruby. For example, we can say object1.extract!(:a, :b). This will output the values assigned to the keys a and b. 
```

Comment:
Great example in your answer, and nice job adding onto your knowledge with research.